components/core/src/clp/streaming_archive/MetadataDB.cpp

```diff
@@ -229,12 +230,22 @@ static SQLitePreparedStatement get_files_select_statement(
     }
 
     // Add ordering
-    fmt::format_to(
-            statement_buffer_ix,
-            " ORDER BY {} ASC, {} ASC",
-            streaming_archive::cMetadataDB::File::SegmentId,
-            streaming_archive::cMetadataDB::File::SegmentTimestampsPosition
-    );
+    if (order_by_end_ts) {
+        fmt::format_to(
+                statement_buffer_ix,
+                " ORDER BY {} DESC, {} ASC, {} ASC",
```

Comment:
If I'm not mistaken, this is going to sort by the files' end timestamp first then by the segment id + timestamp position. This means that if two files, A &B, with non-overlapping time ranges were compressed in the order A, B and their timestamps are ordered, then we will decompress the segment in the reverse order. This will significantly slow down performance when reading from block storage.

Ideally, we want to sort and search the relevant segments in order of their (the segment's) end_timestamp; but within a segment, we should still search files in the order that they are within the segment. Is there a SQL query which can do that?

Reply:
But in this way, we need to scan unnecessary files? Because we want to get latest results, scanning in ascending order may require processing large amount of used data. And can we guarantee the file order in a segment is the same as the timestamp order?

Reply:
Or you mean we scan in order but skip some files based on the metadata?

Reply:
We can't guarantee that all files in a segment have increasing end timestamps, but if they are, since all files in a segment are arranged in a single stream, searching in reverse timestamp order will end up decompressing the same segment multiple times. We could buffer the segment in memory to allow seeking but that's a trade-off we haven't evaluated yet. What I'm suggesting for now is that we search within a segment in byte order (and as you said, skip irrelevant files), but we can choose which segments to search first based on their end timestamps.

Reply:
I don't think we can skip any files unless the query is `*`, but `*` doesn't work well with clo. Can you give me a hint of decompressing the segment multiple times? And from the hive dataset, if we sort the files in timestamp order, their `segment_..._position` are out of order.